### DISCUSSION TITLE
`open` namespace

### DISCUSSION THREAD

**Discussion by oovm** [bytecodealliance/registry#281] [Ideas]
Currently there are two namespaces on wa.dev, `public` and `private`, I hope to add an `open` level

- **open**: everyone can upload, everyone can download
- **public**: members can upload, everyone can download
- **private**: members can upload, members can download

## Motivation

I wrote a language vk, and I want users of the vk language to be able to publish packages with vk metainformation to wa.dev.
I wrote the following rules
- Query all packages in wa.dev when the package name is `@namespace/name`
- When the package name is "name", query `@vk/name`

Therefore I want there to be a `vk` open namespace where all users can upload packages.

If this feature is not supported, I think users may have to write the full name with namespace everywhere, which is very tedious.

**Comment by calvinrp** [comment]
Very cool. Happy to help. If you are OK with it, let's copy this discussion over to the [wa.dev discussions ](https://github.com/JAFLabs/wa.dev-issues/discussions), since it is about `wa.dev` service itself and not the related Bytecode Alliance protocol / tools.

The short answer is we don't currently have a setting for `open` to everyone to publish in a namespace. But we could add it. `public` and `private` are supported today.

There might be things to consider for the `open` namespace. Wondering if there's a better way to solve the lookup problem. We would love to help and support your use case. Let's continue the discussion on our [Discord](https://discord.com/invite/CPb4943knF).

**Comment by lann** [comment]
I'm going to close this discussion as it seems to be entirely wa.dev specific. Please move to https://github.com/JAFLabs/wa.dev-issues as suggested above.
